scanner/config/config_test.go

```diff
@@ -91,4 +91,102 @@ func Test_validate(t *testing.T) {
 		err := c.validate()
 		assert.ErrorContains(t, err, "grpc_listen_addr is empty")
 	})
+	t.Run("when indexer is invalid then error", func(t *testing.T) {
+		c := defaultConfiguration
+		c.Indexer.Database.ConnString = "force indexer to fail validate"
+		err := c.validate()
+		assert.ErrorContains(t, err, "indexer:")
+	})
+	t.Run("when matcher is invalid then error", func(t *testing.T) {
+		c := defaultConfiguration
+		c.Matcher.Database.ConnString = "force matcher to fail validate"
+		err := c.validate()
+		assert.ErrorContains(t, err, "matcher:")
+	})
+}
+
+func Test_IndexerConfig_validate(t *testing.T) {
+	t.Run("when disabled no error", func(t *testing.T) {
+		c := IndexerConfig{Enable: false, Database: Database{ConnString: "invalid conn string"}}
+		err := c.validate()
+		assert.NoError(t, err)
+	})
+	t.Run("when enabled then error", func(t *testing.T) {
```

Comment:
for this and the matcher, can you update this to say "when enabled with invalid conn string then error"?
Otherwise, it looks like just enabling this is an error